aspnet/mvc/models/validation.rst

```diff
@@ -1,8 +1,125 @@
+
+In the following sample, a business rule that states that movies released before 1960 belong to the classic movie genre. The ``[ClassicMovie]`` attribute checks the genre against the release date. If it is released before 1960, genre the property must be categorized as a classic movie. The attribute accepts an integer parameter representing the year that you can use to validate data. You can capture the value of the parameter in the attribute's constructor, as shown here:
+					 
+.. literalinclude:: validation/sample/ClassicMovieAttribute.cs
+   :language: c#
+   :lines: 11-36
+   :dedent: 4
+   
+The ``movie`` variable above represents a ``Movie`` object that contains the data from the form submission to validate. In this case, the validation code checks the date and genre in the ``IsValid`` method of the ``ClassicMovieAttribute`` class as per the rules. Upon successful validation ``isValid`` returns a ``ValidationResult.Success`` code, and when validation fails, a ``ValidationResult`` with an error message. When a user modifies the ``Genre`` field and submits the form, the `IsValid` method of the ``ClassicMovieAttribute`` will verify whether the movie is a classic. Like any built-in attribute, apply the ``ClassicMovieAttribute`` to a property such as ``ReleaseDate`` to ensure validation happens, as shown in the previous code sample. 
+
+Client side validation
+----------------------
+Client side validation is a great convenience for users. It saves time they would otherwise spend waiting for a round trip to the server. In business terms, even a few seconds is multiplied hundreds of times each day and adds up to be a lot of time, expense, and frustration. Straightforward and immediate validation enables users to work more efficiently and produce better quality input and output. 
+
+MVC uses data annotations in addition to type metadata from model properties to validate data and display any error messages. It does so by rendering `HTML5 ``data-`` attributes  <http://w3c.github.io/html/dom.html#embedding-custom-non-visible-data-with-the-data-attributes>`_ in the form elements that need validation, as shown below. MVC generates the ``data-`` attributes for both built-in and custom attributes. The ``data-val-required`` attribute below contains an error message to display if the user doesn't fill in the release date field, and that message displays in the accompanying ``<span>`` element.
```

Comment:
> It does so by rendering `HTML5 ``data-`` attributes

But how do these HTML attributes get rendered? I think you need to say something like "When you use MVC to render form element for a model (using Tag Helpers (link) or HTML helpers (link)) it wil add HTML 5 data- attributes. . ."